ui-speedspacechart/src/types/chartTypes.ts

```diff
@@ -93,6 +93,20 @@ export type DrawFunctionParams = {
   setStore?: React.Dispatch<React.SetStateAction<Store>>;
 };
 
+export type RestrictedStore = Pick<
+  Store,
+  'speedLimitTags' | 'ratioX' | 'leftOffset' | 'layersDisplay' | 'speeds'
+> & {
+  layersDisplay: Pick<Store['layersDisplay'], 'electricalProfiles' | 'powerRestrictions'>;
+};
+
+export type RestrictedDrawFunctionParams = {
+  ctx: CanvasRenderingContext2D;
+  width: number;
+  height: number;
+  store: RestrictedStore;
+};
```

Comment:
Hm, I wonder if "restricted" is a good way to describe this. Is this going to be generally enough for any drawing function to re-use, or is this really specific to drawing speed limit tags, or a specific kind of layer?

Also not a fan of repeating the types… Do we really gain that much by narrowing the type here?

Reply:
I agree with you, the name is a bit imprecise, if I keep it I'll rename it.

The reason I did this was to be able to restrict what I fed to the canvas drawing function. I don't particularly enjoy this workaround either, so maybe I should fully remove the dependence of the drawing function on the store and its associated types and give it each of its arguments independently.

Reply:
To keep some consistency I have renamed and kept the type for now, but I think another pass to reduce the use of the store in all drawing functions could be appreciated. But this should probably be another pr.

Reply:
The new name looks good to me!